### PR TITLE
Migrate widgets to registry

### LINKED ISSUE #3700: Pull chain metadata from registry in Widgets lib
Required for https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3692

### PR DESCRIPTION
**Pull request by jmrossy**
Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3700

- Migrate components and hooks to use Registry utilities
- Remove chain logo components

### CODE REVIEW

File: src/utils/explorers.ts
Comment (yorhodes) [inline src/utils/explorers.ts:1]: nice.

File: src/icons/ChainLogo.tsx [hunk 1]
```diff
@@ -1,114 +1,41 @@
...
 export interface ChainLogoProps {
-  chainId?: number | string;
-  chainName?: string;
+  chainName: string;
+  registry: IRegistry;
   size?: number;
-  color?: boolean;
   background?: boolean;
-  icon?: SvgIcon; // Override the default set used above. Necessary for PI chain logos.
+  Icon?: SvgIcon; // Override the default set used above. Necessary for PI chain logos.
```
Comment (yorhodes) [inline src/icons/ChainLogo.tsx]: I guess we can get rid of this branching once we have Registry overrides?

**Review by yorhodes — APPROVED** [review]
nice, love how minimal the dependency tree is here

**Review by paulbalaji — APPROVED** [review]
nice and clean - loving the nuke too

File: src/icons/ChainLogo.tsx (hunk 1 above)
Comment (jmrossy) [reply src/icons/ChainLogo.tsx]: Yeah it's no longer used much but I can imagine a case or two where we'd want to let someone override the logo so I'll leave it in for now

**Pull request merged**